### PR TITLE
[OTX/Anomaly] Add changes from external to otx

### PR DESCRIPTION
**Pull request by ashwinvaidya17**
# Description

- Adds changes from external/anomaly to otx/algorithms.
- Note: the pre-commit tests are failing on this branch. But I am also getting these errors on the target branch. I think the CI is working now. Maybe we can update the target branch with fixes for pre-commit. I'll update this branch again after that.

### CODE REVIEW

File: otx/algorithms/anomaly/tasks/inference.py [hunk 1]
```diff
@@ -30,6 +30,16 @@
...
+from ote_sdk.entities.datasets import DatasetEntity
+from ote_sdk.entities.inference_parameters import InferenceParameters
+from ote_sdk.entities.metrics import NullPerformance, Performance, ScoreMetric
+from ote_sdk.entities.model import (
+    ModelEntity,
+    ModelFormat,
+    ModelOptimizationType,
+    ModelPrecision,
+    OptimizationMethod,
+)
```
Comment (goodsong81) [inline otx/algorithms/anomaly/tasks/inference.py]: This might be mistake :)

File: otx/algorithms/anomaly/tasks/train.py [hunk 2]
```diff
@@ -14,25 +14,24 @@
...
+from adapters.anomalib.logger import get_logger
+from anomalib.models import AnomalyModule, get_model
 from anomalib.utils.callbacks import (
     MetricsConfigurationCallback,
     MinMaxNormalizationCallback,
 )
+from ote_sdk.entities.datasets import DatasetEntity
+from ote_sdk.entities.model import ModelEntity
+from ote_sdk.entities.train_parameters import TrainParameters
+from ote_sdk.usecases.tasks.interfaces.training_interface import ITrainingTask
```
Comment (goodsong81) [inline otx/algorithms/anomaly/tasks/train.py]: Same here.

File: otx/algorithms/anomaly/tasks/train.py [hunk 3]
```diff
@@ -68,11 +66,10 @@ def train(
 
         logger.info("Training Configs '%s'", config)
 
-        datamodule = OTXAnomalyDataModule(config=config, dataset=dataset, task_type=self.task_type)
+        datamodule = OTEAnomalyDataModule(config=config, dataset=dataset, task_type=self.task_type)
```
Comment (goodsong81) [inline otx/algorithms/anomaly/tasks/train.py]: -> OTX

**Review by goodsong81 — CHANGES_REQUESTED** [review]
Thank you for the PR. Could you revise a little bit?
Don't mind the code check. I've fixed them. (You may want to rebase)

**Comment by ashwinvaidya17** [comment]
Anomaly tests 
image

**Comment by ashwinvaidya17** [comment]
@goodsong81 thanks for the review! I missed those. I've updated the PR.

**Review by goodsong81 — APPROVED** [review]
LGTM, thank you!
Would the DRAEM be dealt with separately?
Or, no need to be applied to feature/otx?

File: otx/algorithms/anomaly/configs/base/configuration.py [hunk 4]
```diff
@@ -48,6 +49,16 @@ class LearningParameters(ParameterGroup):
         header = string_attribute("Learning Parameters")
         description = header
 
+        # Editable is set to false as WideResNet50 is very large for
+        # onnx's protobuf (2gb) limit. This ends up crashing the export.
+        backbone = selectable(
```
Comment (djdameln) [inline otx/algorithms/anomaly/configs/base/configuration.py]: if the configurable parameters are changed, should the `configuration.yaml` files be updated as well?

**Review by djdameln — COMMENTED** [review]
I have the same question as Songki, did you take DRAEM into consideration? 

By adding the `backbone` parameter to the base config, that parameter is now exposed to all models, including DRAEM. But DRAEM does not support different backbones, so this parameter should not be exposed.

File: otx/algorithms/anomaly/configs/base/configuration.py (hunk 4 above)
Comment (ashwinvaidya17) [reply otx/algorithms/anomaly/configs/base/configuration.py]: I think I was looking at a different PR for the diff. I have removed backbone from configuration.py and added the changes for nncf in configuration.py. But apart from that I don't think there are other changes between external and otx for DRAEM.

File: otx/algorithms/anomaly/configs/base/configuration.py [hunk 5]
```diff
@@ -19,6 +19,7 @@
 from attr import attrs
 
 from otx.algorithms.anomaly.configs.base.configuration_enums import (
+    ModelBackbone,
```
Comment (djdameln) [inline otx/algorithms/anomaly/configs/base/configuration.py]: I guess we don't need to import this if it's not used

Comment (goodsong81) [reply otx/algorithms/anomaly/configs/base/configuration.py]: I'll take care. Thanks!

**Pull request merged**